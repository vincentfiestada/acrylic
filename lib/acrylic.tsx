/**
 * Re-export all components in one place
 */
export { default as IProps } from "./components/IProps";
export { default as IEventProps } from "./components/IEventProps";

/**
 * Layout Components
 */
export { default as ILayoutProps } from "./components/layout/ILayoutProps";
export { default as Section } from "./components/layout/Section";
export { default as Column } from "./components/layout/Column";
export { default as Row } from "./components/layout/Row";
export { default as Alert } from "./components/layout/Alert";
export { default as Toolbar } from "./components/layout/Toolbar";

/**
 * Typography Components
 */
export { default as ITextProps } from "./components/typography/ITextProps";
export { default as Title } from "./components/typography/Title";
export { default as Text } from "./components/typography/Text";
export { default as Icon } from "./components/typography/Icon";

/**
 * Control Components
 */
export { default as Button } from "./components/controls/Button";
export { default as IconButton } from "./components/controls/IconButton";

/**
 * Variants
 */
export { default as Variant } from "./components/Variant";
