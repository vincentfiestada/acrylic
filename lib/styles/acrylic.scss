@import "normalize.css";
@import "typography";
@import "controls";
