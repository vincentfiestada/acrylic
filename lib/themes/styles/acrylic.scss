@import "normalize.css";
@import "typography";
@import "icons";
@import "layout";
@import "controls";
@import "progress";
@import "animation";
