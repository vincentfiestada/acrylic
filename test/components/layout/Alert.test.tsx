import * as React from "react";
import { mount } from "enzyme";
import { Alert, Button } from "../../../lib/acrylic";

describe("<Alert />", () => {
	it("Should match the latest snapshot", () => {
		expect(
			mount(
				<React.Fragment>
					<Alert>Example</Alert>
					<Alert variant="primary">Example</Alert>
					<Alert variant="success">Example</Alert>
					<Alert variant="info">Example</Alert>
					<Alert variant="warning">Example</Alert>
					<Alert variant="danger">Example</Alert>
					<Alert controlComponent={() => <Button>Control</Button>}>
						Example
					</Alert>
				</React.Fragment>
			)
		).toMatchSnapshot();
	});
	it("Should be dismissible", () => {
		const alertComponent = mount(<Alert>Example</Alert>);
		alertComponent
			.find(Button)
			.at(0)
			.simulate("click");
		expect(alertComponent.hasClass("acr-hidden"));
	});
});
