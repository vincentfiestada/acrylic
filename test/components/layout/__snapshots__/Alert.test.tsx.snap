// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Should match the latest snapshot 1`] = `
Array [
  <Alert>
    <aside
      className="acr-alert"
    >
      Example
      <Button
        onClick={[Function]}
      >
        <button
          className="acr-btn"
          onClick={[Function]}
        >
          Dismiss
        </button>
      </Button>
    </aside>
  </Alert>,
  <Alert
    variant="primary"
  >
    <aside
      className="acr-alert acr-variant-primary"
    >
      Example
      <Button
        onClick={[Function]}
        variant="primary"
      >
        <button
          className="acr-btn acr-variant-primary"
          onClick={[Function]}
        >
          Dismiss
        </button>
      </Button>
    </aside>
  </Alert>,
  <Alert
    variant="success"
  >
    <aside
      className="acr-alert acr-variant-success"
    >
      Example
      <Button
        onClick={[Function]}
        variant="success"
      >
        <button
          className="acr-btn acr-variant-success"
          onClick={[Function]}
        >
          Dismiss
        </button>
      </Button>
    </aside>
  </Alert>,
  <Alert
    variant="warning"
  >
    <aside
      className="acr-alert acr-variant-warning"
    >
      Example
      <Button
        onClick={[Function]}
        variant="warning"
      >
        <button
          className="acr-btn acr-variant-warning"
          onClick={[Function]}
        >
          Dismiss
        </button>
      </Button>
    </aside>
  </Alert>,
  <Alert
    variant="danger"
  >
    <aside
      className="acr-alert acr-variant-danger"
    >
      Example
      <Button
        onClick={[Function]}
        variant="danger"
      >
        <button
          className="acr-btn acr-variant-danger"
          onClick={[Function]}
        >
          Dismiss
        </button>
      </Button>
    </aside>
  </Alert>,
  <Alert>
    <aside
      className="acr-alert"
    >
      Example
      <Button
        onClick={[Function]}
      >
        <button
          className="acr-btn"
          onClick={[Function]}
        >
          Control
        </button>
      </Button>
    </aside>
  </Alert>,
]
`;
